---
layout: post
title: "Missing data and PCA"
subtitle: "How much is too much?"
authors:
 - Thiseas C. Lamnidis
categories: Blog
tag: Blog
---

_by [Thiseas C. Lamnidis](https://www.eva.mpg.de/archaeogenetics/staff/thiseas-christos-lamnidis/)_

## Introduction

In the field of ancient human population genetics, we often analyse ancient genomes with high proportions of missing data, namely in the recovered genotypes.
This can stem from limitations of material preservation, and/or of available laboratory/sequencing technologies.
Given that sampling remains for ancient DNA is a destructive process, we have an ethical responsibility to make the most of the data we are able to generate.
At the same time, we need to know exactly the limitations of our data when inferring results, and present our findings fairly and compassionately.
After all, the past breeds the present, and the interpretations we make of the past can have implications affecting present-day peoples and communities.

To that end, archaeogenticists often use quality cutoff thresholds in our analyses, limiting the set of analysed individuals to ones with sufficient SNP coverage for population genetics analyses.
These cutoffs are often expressed in terms of the "Number of covered SNPs", i.e. the number of SNPs which do not have a missing genotype call for a given ancient individual.
But what constitutes a sensible minimum cutoff?

Principal Component Analysis (PCA) is a very useful _visualisation_ tool for human aDNA data, as well as the source of many-a [popgen bingo](https://ancientpopgen-bingo.firebaseapp.com/?game=ancientpopgen-bingo).
It is often the case that most (if not all) sampled individuals will be featured in a PCA plot, and it is important to understand how missing data affects their position in order to avoid misinterpreting PCA results.
So, I set out to find out:

How does missing data affect the projected position of individuals on a PCA plot?
Can you trust the placement of that low coverage individual in the paper you're reviewing, and if so to what degree?

Let's find out!

## Experimental setup

I started with a dataset of present day individuals on the Human Origins array (\~600k SNPs) <sup>[[1](http://dx.doi.org/10.1534/genetics.112.145037),[2](http://dx.doi.org/10.1038/nature13673),[3](https://doi.org/10.1038/s41559-019-0878-2)]</sup>.
I picked a set of candidate present-day individuals (plus the 'Loschbour' hunter-gatherer) whose positions roughly spanned the area of each PCA plot.
I then randomly replaced the existing genotypes with missing ones for each candidate individual at a given rate, creating 200 replicates per missingness rate, per individual.
I used smartpca <sup>[[4](http://dx.doi.org/10.1371/journal.pgen.0020190)]</sup> to run PCA and projected all replicates (lsqproject: YES) across all missingness rates for all candidate individuals.
The original candidate individuals were not projected, with the exception of 'Loschbour', which is also the only individual with pseudohaploid genotypes in the experiment.
The 'Loschbour' hunter-gatherer can then act as a test case for true ancient data, where differences between the replicate spread of this individual to other candidate individuals could indicate that damage, genotyping procedure, and/or projecting individuals whose genetic variation is not directly included in the PC calculation can also affect the spread of data due to missingness.

I focused on four PCA setups that are somewhat prevalent in human aDNA literature:

-   Worldwide PCA

-   Eurasian PCA (à la [Jeong2019](https://doi.org/10.1038/s41559-019-0878-2))

-   West Eurasian PCA (à la [Lazaridis et al. 2014](http://dx.doi.org/10.1038/nature13673))

-   African PCA (à la [Skoglund et. al 2017](http://dx.doi.org/10.1016/j.cell.2017.08.049))

## Results

In each slideshow below, you can see the results for eahc of the regional PCAs for missingness rates of 90%, 97%, and 99%, corresponding to approximately 60.000, 18.000, and 6.000 covered SNPs resprectively.
Present-day populations are shown in the background, plotted as a unique 3-4 letter code per population and coloured according to their approximate geographical location/region.
The "true" position of each candidate individual (i.e. their position when using all the available genotypes) is shown by black symbols.
Around each of these symbols, the convex hull of the 200 replicates with artificailly-introduced missing genotypes is plotted, representing the variability of the projected placement of the replicates individuals.

### Worldwide

<!-- Slideshow container -->
<div class="slideshow-container">

  <!-- Full-width images with number and caption text -->
  <div class="mySlides slide0 fade">
    <div class="numbertext">1 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/Worldwide.m0.9.png" | relative_url }}" style="width:90%">
    <br>
    <div class="text">Spread of replicates with 90% of genotypes replaced by missing data.</div>
  </div>

  <div class="mySlides slide0 fade">
    <div class="numbertext">2 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/Worldwide.m0.97.png" | relative_url }}" style="width:90%">
   <br>
    <div class="text">Spread of replicates with 97% of genotypes replaced by missing data.</div>
  </div>

  <div class="mySlides slide0 fade">
    <div class="numbertext">3 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/Worldwide.m0.99.png" | relative_url }}" style="width:90%">
   <br>
    <div class="text">Spread of replicates with 99% of genotypes replaced by missing data.</div>
  </div>

  <!-- Next and previous buttons -->
  <a class="prev" onclick="plusSlides(-1, 0)">&#10094;</a>
  <a class="next" onclick="plusSlides(1, 0)">&#10095;</a>
</div>
<br>

### Eurasia

<!-- Slideshow container -->
<div class="slideshow-container">

  <!-- Full-width images with number and caption text -->
  <div class="mySlides slide1 fade">
    <div class="numbertext">1 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/Eurasia.m0.9.png" | relative_url }}" style="width:90%">
    <br>
    <div class="text">Spread of replicates with 90% of genotypes replaced by missing data.</div>
  </div>

  <div class="mySlides slide1 fade">
    <div class="numbertext">2 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/Eurasia.m0.97.png" | relative_url }}" style="width:90%">
   <br>
    <div class="text">Spread of replicates with 97% of genotypes replaced by missing data.</div>
  </div>

  <div class="mySlides slide1 fade">
    <div class="numbertext">3 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/Eurasia.m0.99.png" | relative_url }}" style="width:90%">
   <br>
    <div class="text">Spread of replicates with 99% of genotypes replaced by missing data.</div>
  </div>

  <!-- Next and previous buttons -->
  <a class="prev" onclick="plusSlides(-1, 1)">&#10094;</a>
  <a class="next" onclick="plusSlides(1, 1)">&#10095;</a>
</div>
<br>

### West Eurasia

<!-- Slideshow container -->
<div class="slideshow-container">

  <!-- Full-width images with number and caption text -->
  <div class="mySlides slide2 fade">
    <div class="numbertext">1 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.9.png" | relative_url }}" style="width:90%">
    <br>
    <div class="text">Spread of replicates with 90% of genotypes replaced by missing data.</div>
  </div>

  <div class="mySlides slide2 fade">
    <div class="numbertext">2 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.97.png" | relative_url }}" style="width:90%">
   <br>
    <div class="text">Spread of replicates with 97% of genotypes replaced by missing data.</div>
  </div>

  <div class="mySlides slide2 fade">
    <div class="numbertext">3 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.99.png" | relative_url }}" style="width:90%">
   <br>
    <div class="text">Spread of replicates with 99% of genotypes replaced by missing data.</div>
  </div>

  <!-- Next and previous buttons -->
  <a class="prev" onclick="plusSlides(-1, 2)">&#10094;</a>
  <a class="next" onclick="plusSlides(1, 2)">&#10095;</a>
</div>
<br>

### Africa

<!-- Slideshow container -->
<div class="slideshow-container">

  <!-- Full-width images with number and caption text -->
  <div class="mySlides slide3 fade">
    <div class="numbertext">1 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/Africa.m0.9.png" | relative_url }}" style="width:90%">
    <br>
    <div class="text">Spread of replicates with 90% of genotypes replaced by missing data.</div>
  </div>

  <div class="mySlides slide3 fade">
    <div class="numbertext">2 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/Africa.m0.97.png" | relative_url }}" style="width:90%">
   <br>
    <div class="text">Spread of replicates with 97% of genotypes replaced by missing data.</div>
  </div>

  <div class="mySlides slide3 fade">
    <div class="numbertext">3 / 3</div>
    <img src="{{ "/assets/media/event_images/2024-04-14-blog/Africa.m0.99.png" | relative_url }}" style="width:90%">
   <br>
    <div class="text">Spread of replicates with 99% of genotypes replaced by missing data.</div>
  </div>

  <!-- Next and previous buttons -->
  <a class="prev" onclick="plusSlides(-1, 3)">&#10094;</a>
  <a class="next" onclick="plusSlides(1, 3)">&#10095;</a>
</div>
<br>

## Conclusions

In literature, the chosen cutoffs are both sensible and (strictly speaking) arbitrary, commonly ranging between 10.000 and 20.000 covered SNPs.
Any individuals with fewer than that number of SNPs are often excluded from downstream analyses.
Looking at the results above, you can see why:

 - High degrees of missingness can affect the projected placement of an individual in regional PCAs. 
    - e.g. with sparse enough data (99% missing), some replicates of the Sicilian individual 'TP08' are projected closer to present-day Greeks than the other Sicilians in the dataset.
 - While the exact projection of low-coverage individuals in the Worldwide PCA is also affected, the spread of replicates shows that it is still perfectly possible to assess if a studied individual is genetically more closely related to present-day African, South Asian or Native American populations from the PCA results, even from only a few thousand SNPs.
 - Comparing the spread of replicates around 'Loschbour' to those around the other candidate individuals, it does not look like projection of the candidate individual itself affects the spread of the replicates. (n=1!!)

Hopefully, these results help you gauge if and how much the position of your low coverage individuals on your favourite PCA can be trusted.

### Caveats

The experiment preformed here assumes that missingness is random, but it is conceivable that not every SNP has the same probability of getting lost through degradation processes.

The hulls shown here are specifically for the Human Origins dataset.
The ascertainment of each dataset could alter the effects of missingness, depending also on how the PCs were calculated.


<!-- CSS for plot slideshow. See: https://www.w3schools.com/howto/howto_js_slideshow.asp-->
<style>
* {box-sizing:border-box}
img {vertical-align: middle;}

/* Slideshow container */
.slideshow-container {
  max-width: 1000px;
  position: relative;
  margin: auto;
}

/* Hide the images by default */
.mySlides {
  display: none;
}

/* Next & previous buttons */
.prev, .next {
  cursor: pointer;
  position: absolute;
  top: 50%;
  width: auto;
  margin-top: -22px;
  padding: 16px;
  color:  #3A5FCD;
  font-weight: bold;
  font-size: 18px;
  transition: 0.6s ease;
  border-radius: 0 3px 3px 0;
  user-select: none;
}

/* Position the "next button" to the right */
.next {
  right: 0;
  border-radius: 3px 0 0 3px;
}

/* On hover, add a black background color with a little bit see-through */
.prev:hover, .next:hover {
  background-color: rgba(0,0,0,0.8);
}

/* Caption text */
.text {
  color: black;
  font-size: 15px;
  padding: 0px 12px;
  position: absolute;
  bottom: 8px;
  width: 100%;
  text-align: center;
}

/* Number text (1/3 etc) */
.numbertext {
  color: #f2f2f2;
  font-size: 12px;
  padding: 8px 12px;
  position: absolute;
  top: 0;
}

/* The dots/bullets/indicators */
.dot {
  cursor: pointer;
  height: 15px;
  width: 15px;
  margin: 0 2px;
  background-color: #bbb;
  border-radius: 50%;
  display: inline-block;
  transition: background-color 0.6s ease;
}

.active, .dot:hover {
  background-color: #717171;
}

/* Fading animation */
.fade {
  animation-name: fade;
  animation-duration: 1.5s;
}

@keyframes fade {
  from {opacity: .4}
  to {opacity: 1}
}
</style>

<!-- Javascript for plot slideshows. See: https://www.w3schools.com/howto/howto_js_slideshow.asp-->
<script>
var slideIndex = [1,1,1,1];
var slideId = ["slide0", "slide1", "slide2", "slide3"]
showSlides(1, 0);
showSlides(1, 1);
showSlides(1, 2);
showSlides(1, 3);

function plusSlides(n, no) {
  showSlides(slideIndex[no] += n, no);
}

function showSlides(n, no) {
  var i;
  var x = document.getElementsByClassName(slideId[no]);
  if (n > x.length) {slideIndex[no] = 1}    
  if (n < 1) {slideIndex[no] = x.length}
  for (i = 0; i < x.length; i++) {
     x[i].style.display = "none";  
  }
  x[slideIndex[no]-1].style.display = "block";  
}
</script>