---
layout: post
title: "Missing data and PCA"
subtitle: "How much is too much?"
authors:
 - Thiseas C. Lamnidis
categories: Blog
tag: Blog
---

_by [Thiseas C. Lamnidis](https://www.eva.mpg.de/archaeogenetics/staff/thiseas-christos-lamnidis/)_

## Introduction

In the field of ancient human population genetics, we often analyse ancient genomes with high proportions of missing genotypes.
This can stem from limitations of material preservation, and/or of available laboratory/sequencing technologies.
Given that sampling remains for ancient DNA is a destructive process, we have the ethical responsibility to make the most of the data we are able to generate.
Additionally, we need to know exactly the limitations of our data when inferring results, and present our findings fairly and compassionately.
After all, the past breeds the present, and the interpretations we make of the past can have implications affecting present-day peoples and communities.

To that end, archaeogenticists often use quality cutoff thresholds in our analyses, limiting the set of analysed individuals to ones with sufficient coverage for population genetics analyses.
These cutoffs are often expressed in terms of the "Number of covered SNPs", i.e. the number of SNPs which do not have a missing genotype call for a given ancient individual.
But what constitutes a sensible minimum cutoff?

It is often the case that most (if not all) sampled individuals will be featured in a Principal Component Analysis (PCA), which is both a very useful visualisation tool for human aDNA data, and the source of many-a [popgen bingo](https://ancientpopgen-bingo.firebaseapp.com/?game=ancientpopgen-bingo).
How does missing data affect the projected position of ancient individuals on a PCA plot?
Can you trust the placement of that low coverage individual in the paper you're reviewing, and if so to what degree?

Let's find out!

## Experimental setup

I started with a dataset of present day individuals on the Human Origins array (\~600k SNPs) <sup>[[1](http://dx.doi.org/10.1534/genetics.112.145037),[2](http://dx.doi.org/10.1038/nature13673),[3](https://doi.org/10.1038/s41559-019-0878-2)]</sup>.
I picked a set of candidate present-day individuals (plus the 'Loschbour' hunter-gatherer) whose positions roughly spanned the area of each PCA plot.
I then randomly added missing data to each candidate individual's genotypes at a given rate, creating 200 replicates per missingness rate, per individual.
I then used smartpca [4](http://dx.doi.org/10.1371/journal.pgen.0020190) to run PCA and projected all replicates (`lsqproject: YES`) across all missingness rates for all candidate individuals. The original candidate individuals were not projected, with the exception of 'Loschbour', which is also the only individual with pseudohaploid genotypes in the experiment.

I focused on four PCA setups that are somewhat prevalent in human aDNA literature:

-   Worldwide PCA

-   Eurasian PCA (à la [Jeong2019](https://doi.org/10.1038/s41559-019-0878-2))

-   West Eurasian PCA (à la [Lazaridis et al. 2014](http://dx.doi.org/10.1038/nature13673))

-   African PCA (à la [Skoglund et. al 2017](http://dx.doi.org/10.1016/j.cell.2017.08.049))

## Results

<sup>(You can right click on any of the images below and select "Open Image in a New Tab" to see them in more detail.)</sup>

### Missing 90% of all genotypes (60.000 SNPs)

| :-- | --: |
![Worldwide](/assets/media/event_images/2024-04-14-blog/Worldwide.m0.9.png)| ![Eurasian](/assets/media/event_images/2024-04-14-blog/Eurasia.m0.9.png)
![West Eurasian](/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.9.png)|![African](/assets/media/event_images/2024-04-14-blog/Africa.m0.9.png)

### Missing 97% of all genotypes (18.000 SNPs)

| :-- | --: |
![Worldwide](/assets/media/event_images/2024-04-14-blog/Worldwide.m0.97.png) |![Eurasian](/assets/media/event_images/2024-04-14-blog/Eurasia.m0.97.png)
![West Eurasian](/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.97.png) |![African](/assets/media/event_images/2024-04-14-blog/Africa.m0.97.png)

### Missing 99% of all genotypes (6.000 SNPs)

| :-- | --: |
![Worldwide](/assets/media/event_images/2024-04-14-blog/Worldwide.m0.99.png) |![West Eurasian](/assets/media/event_images/2024-04-14-blog/WestEurasia.m0.99.png)

## Conclusions

In literature, the chosen cutoffs are both sensible and (strictly speaking) arbitrary, commonly ranging between 10.000 and 20.000 covered SNPs.
Any individuals with fewer than that number of SNPs are often excluded from downstream analyses.
Looking at the results above, you can see why:

 - High degrees of missingness can affect the projected placement of an individual in regional PCAs. 
    - e.g. with sparse enough data (99% missing), some replicates of the Sicilian individual 'TP08' are projected closer to present-day Greeks than the other Sicilians in the dataset.
 - While the exact projection of low-coverage individuals in the Worldwide PCA is also affected, the spread of replicates shows that it is still perfectly possible to assess if a studied individual is genetically more closely related to present-day African, South Asian or Native American populations from the PCA results, even from only a few thousand SNPs.
 - Comparing the spread of replicates around 'Loschbour' to those around the other candidate individuals, it does not look like projection of the candidate individual itself affects the spread of the replicates. (n=1!!)

Hopefully, these results help you gauge if and how much the position of your low coverage individuals on your favourite PCA can be trusted.

### Caveats

The experiment preformed here assumes that missingness is random, but it is conceivable that not every SNP has the same probability of getting lost through degradation processes.

The hulls shown here are specifically for the Human Origins dataset.
The ascertainment of each dataset could alter the effects of missingness, depending also on how the PCs were calculated.